contracts/nois-proxy/src/msg.rs

```diff
@@ -146,6 +161,11 @@ pub struct GatewayChannelResponse {
     pub channel: Option<String>,
 }
 
+#[cw_serde]
+pub struct IsAllowedResponse {
+    pub is_allowed: bool,
```

Comment:
```suggestion
    /// Returns true if and only if the address is allowed to use the proxy.
    /// This is true if no allow-list is in place or the given address is allow-listed.
    pub is_allowed: bool,
```

Reply:
Actually I am not sure how the query should work. I think what you implemented is simpler (just the existence in the allowlist). If the list is enabled can be checked separately.